test/e2e/framework/util.go

```diff
@@ -2350,15 +2350,27 @@ func waitListSchedulableNodesOrDie(c clientset.Interface) *api.NodeList {
 	return nodes
 }
 
+// isNodeTainted checks if a node is tainted and thus not schedulable to pods (without a toleration)
+func isNodeTainted(node *api.Node) bool {
+	nodeTaints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		Failf("error getting taints: %v", err)
+	}
+
+	return len(nodeTaints) > 0
+}
+
 // Node is schedulable if:
 // 1) doesn't have "unschedulable" field set
 // 2) it's Ready condition is set to true
 // 3) doesn't have NetworkUnavailable condition set to true
+// 4) it isn't tainted
 func isNodeSchedulable(node *api.Node) bool {
 	nodeReady := IsNodeConditionSetAsExpected(node, api.NodeReady, true)
+	isTainted := isNodeTainted(node)
 	networkReady := IsNodeConditionUnset(node, api.NodeNetworkUnavailable) ||
 		IsNodeConditionSetAsExpectedSilent(node, api.NodeNetworkUnavailable, false)
-	return !node.Spec.Unschedulable && nodeReady && networkReady
+	return !node.Spec.Unschedulable && nodeReady && networkReady && !isTainted
```

Comment:
this might happen to work now, but taints/tolerations really makes it impossible to say that a node is schedulable or unschedulable in isolation -- you have to look at the pod in question also, and say the node is unschedulable *for that pod* iff the node has a taint that the pod does not tolerate. For example see PodToleratesNodeTaints() in plugin/pkg/scheduler/algorithm/predicates/predicates.go. (Of course the other tests apply in addition, like nodeReady, networkReady, etc. that is earlier in this line)